### PR TITLE
Test that handling to device messages off master works

### PR DESCRIPTION
**Pull request by erikjohnston**
Test for https://github.com/matrix-org/synapse/pull/9044

### CODE REVIEW

File: lib/SyTest/Homeserver/Synapse.pm [hunk 1]
```diff
@@ -307,10 +307,15 @@ sub start
...
            },
+	   "client_reader" => {
+              host => "$bind_host",
+              port => $self->{ports}{client_reader},
+           },
         },
 
         stream_writers => {
            events => $self->{redis_host} ne '' ? [ "event_persister1", "event_persister2" ] : "master",
+	   to_device => $self->{redis_host} ne '' ? [ "client_reader" ] : "master",
```
Comment (erikjohnston) [inline lib/SyTest/Homeserver/Synapse.pm:318]: There's no particular reason to choose `client_reader`, but I couldn't think of a better place and I'm not sure we want to add more workers at this point

File: lib/SyTest/Homeserver/Synapse.pm [hunk 2]
```diff
@@ -307,10 +307,15 @@ sub start
               host => "$bind_host",
               port => $self->{ports}{event_persister2},
            },
+	   "client_reader" => {
```
Comment (richvdh) [inline lib/SyTest/Homeserver/Synapse.pm:310]:
```suggestion
	    "client_reader" => {
```

File: lib/SyTest/Homeserver/Synapse.pm (hunk 1 above)
Comment (richvdh) [inline lib/SyTest/Homeserver/Synapse.pm:318]:
```suggestion
	     to_device => $self->{redis_host} ne '' ? [ "client_reader" ] : "master",
```

Comment (richvdh) [reply lib/SyTest/Homeserver/Synapse.pm:318]: maybe stick those words in a comment

**Review by richvdh — APPROVED** [review]
looks fine otherwise

File: lib/SyTest/Homeserver/Synapse.pm (hunk 1 above)
Comment (erikjohnston) [reply lib/SyTest/Homeserver/Synapse.pm]: Done :)

**Pull request merged**